### PR TITLE
Preferences: Add option to enable/disable auto translation on launch

### LINKED ISSUE #139: Add options to configure define default input/output language
I think it will be nice to have an option to define input & output language via command line options and in settings.

Example:
```flatpak run com.github.gi_lom.dialect -t "How are you?" -f english -o french```

+ new options in settings.

Otherwise, it tends to use 'auto' input language and it forces user to pick one manually.
Screenshot from 2021-03-26 10-46-37

### PR DESCRIPTION
**Pull request by rafaelmardojai**
Fixes #139

### CODE REVIEW

File: data/resources/preferences.ui [hunk 1]
```diff
@@ -56,6 +56,21 @@
                 <property name="visible">True</property>
               </object>
             </child>
+            <child>
+              <object class="HdyActionRow">
+                <property name="title" translatable="yes">Auto translation on startup</property>
```
Comment (rafaelmardojai) [inline data/resources/preferences.ui:61]: Any ideas for the text here?

File: src/preferences.py [hunk 2]
```diff
@@ -72,6 +73,8 @@ def setup(self):
                            Gio.SettingsBindFlags.DEFAULT)
         self.settings.bind('translate-accel', self.translate_accel,
                            'selected-index', Gio.SettingsBindFlags.DEFAULT)
+        self.settings.bind('auto-trans', self.auto_trans, 'active',
```
Comment (rafaelmardojai) [inline src/preferences.py:76]: I have a crash sometimes when switching the preference

**Comment by rafaelmardojai** [comment]
I have added command line options to define languages, but haven't time to test it in real world.

File: src/preferences.py (hunk 2 above)
Comment (fushinari) [reply src/preferences.py:76]: Yeah, I reproduced it twice too but now that I'm ready to debug it, it just won't crash.

File: src/main.py [hunk 3]
```diff
@@ -46,19 +51,29 @@ def do_activate(self):
...
+                langs=self.launch_langs,
                 settings=self.settings
             )
         self.window.present()
 
     def do_command_line(self, command_line):
         options = command_line.get_options_dict()
         options = options.end().unpack()
+        text = ''
+        langs = [None, None]
```
Comment (fushinari) [inline src/main.py:63]: Probably should use a dict here for readability.

File: data/resources/preferences.ui (hunk 1 above)
Comment (fushinari) [reply data/resources/preferences.ui:61]: `Use "Auto" as default language on launch` maybe?

File: src/main.py (hunk 3 above)
Comment (rafaelmardojai) [reply src/main.py]: Done!

File: data/resources/preferences.ui (hunk 1 above)
Comment (rafaelmardojai) [reply data/resources/preferences.ui]: That sounds beter, at least for now.

Comment (rafaelmardojai) [reply data/resources/preferences.ui]: That sounds better, at least for now.

File: src/preferences.py (hunk 2 above)
Comment (rafaelmardojai) [reply src/preferences.py:76]: It seems to only happen when I open the preference window a second time.

Comment (rafaelmardojai) [reply src/preferences.py:76]: And I can reproduce it with all the others switches.

Comment (fushinari) [reply src/preferences.py]: Looks like a GTK bug or something, I don't think we're doing anything unconventional here.

Comment (rafaelmardojai) [reply src/preferences.py]: Just for the record, it only happens with `Gio.Settings.bind`, when you save the setting manually works fine.

Comment (rafaelmardojai) [reply src/preferences.py]: Calling `Gio.Settings.unbind` when `DialectPreferencesWindow` is destroyed fixed the issue. 

So looks like the preference controls are not being finalized when `DialectPreferencesWindow` is destroyed.

From  `Gio.Settings.unbind`  docs:

> Note that bindings are automatically removed when the object is finalized, so it is rarely necessary to call this function.

Comment (rafaelmardojai) [reply src/preferences.py]: Funny thing, it only happens when binding `GtkSwitch` but not with `HdyComboRow`.

Comment (fushinari) [reply src/preferences.py]: So, GtkSwitch is not finalizing then?

> Just for the record, it only happens with Gio.Settings.bind, when you save the setting manually it works fine.

Yes, I'd seen this too and was rewriting it to be completely manual (even though that's not ideal). Glad to see there's another way.

Comment (fushinari) [reply src/preferences.py]: Can confirm. The issue has been fixed. Still kinda weird that we have to manually unbind though...

**Pull request merged**